spec/puppet-server-lib/puppet/jvm/execution_spec.rb

```diff
@@ -2,43 +2,63 @@
 
 require 'puppet/server/execution'
 
+def test_execute(command, args = nil, options = {})
+  if args.nil?
+    orig_command_str = command
+  else
+    orig_command_str = "#{command} #{args.join(" ")}"
+  end
+  # NOTE: the `execute` function is not part of the public API,
+  #  but is still the best level of abstraction to use for this sort
+  #  of test, so we're abusing Ruby's `send` for these tests.
+  Puppet::Server::Execution.send(:execute,
```

Comment:
I added `private` to the production code to try to enforce that this is not public API.